pypreprocess/io_utils.py

```diff
@@ -959,3 +959,46 @@ def sanitize_fwhm(fwhm):
             raise ValueError("fwhm must be float or list of 3 "
                              "floats; got %s" % fwhm)
     return fwhm
+
+
+def nii2niigz(ifilename, output_dir=None):
+    """
+    Converts .nii to .nii.gz
+
+    Parameters
+    ----------
+    ifilename: string, of list of strings
+        input filename of image to be compressed
+
+    output_dir: string, optional (default None)
+        output directory to which compressed file will be written.
+        If no value is given, the output will be written in the parent
+        directory of ifilename
+
+    Returns
+    -------
+    ofilename: string
+        filename of extracted image
+
+    """
+
+    if isinstance(ifilename, list):
+        return [nii2niigz(x, output_dir=output_dir) for x in ifilename]
+    else:
+        if not isinstance(ifilename, basestring):
+            raise RuntimeError(
+                "ifilename must be string or list of strings, got %s" % type(
+                    ifilename))
+
+    if not ifilename.endswith('.nii'):
+        return ifilename
+
+    ofilename = ifilename + '.gz'
+    if output_dir is not None:
```

Comment:
if `output_dir` is `None`, you should compute it as
`output_dir = os.path.dirname(input_filename)`